packages/hashi/src/mainClient.js

```diff
@@ -104,6 +120,94 @@ export default class MainClient {
       });
     });
   }
+
+  __fetchContentData(message) {
+    // based on the incoming information, get data
+    // from Kolibri to pass back to the iframe
+
+    // if filtering by optional params
+    if (message.options) {
+      let getParams = {};
+      let options = message.options;
+      if (options.parent && options.parent == 'self') {
+        // need to fetch this value when this
+        // function is move to a location that has access
+        // to `content`
+        // getParams.parent = rootNode;
+      } else if (options.parent) {
+        getParams.parent = options.parent;
+      }
+      options.ids ? (getParams.ids = options.ids) : null;
+      options.page ? (getParams.page = options.page) : null;
+      options.pageSize ? (getParams.ids = options.pageSize) : null;
+      ContentNodeResource.fetchCollection({ getParams }).then(contentNodes => {
+        contentNodes ? (message.status = 'success') : (message.status = 'failure');
+        let response = {};
+        response.page = message.options.page ? message.options.page : 1;
+        response.pageSize = message.options.pageSize ? message.options.pageSize : 50;
+        response.results = contentNodes;
+        message.data = response;
+        message.type = 'response';
+        this.mediator.sendMessage({
+          nameSpace,
+          event: events.DATARETURNED,
+          data: message,
+        });
+      });
+    }
+    // or, if getting by a specific id
+    else if (message.id) {
```

Comment:
As there are no shared code paths between these two parts of the function - might be good to just break them up into two functions, you can then add the conditionalization when you setup the listener above:
```
    this.on(this.events.DATAREQUESTED, message => {
     if (message.dataType === 'Collection') {
      this.__fetchContentCollection(message);
    else if (message.dataType === 'Model')
      this.__fetchContentModel(message);
    }
    });
```

Also suggesting here to add an explicit data type - just to be explicit about what is requested.